### ISSUE TITLE
Stop token can be removed for simplification

### ISSUE THREAD

**Issue by begeekmyfriend** [Rayhane-mamah/Tacotron-2#46]
I do not think we have to train extra stop token targets to learn when to stop decoding. We can replace this [line](https://github.com/Rayhane-mamah/Tacotron-2/blob/master/tacotron/models/helpers.py#L108) with as follows:
```python
finished = tf.reduce_all(tf.equal(self._targets[:, time], [_target_pad]), axis=1)
```
It would seek the `_target_pad` as the terminating symbol to set `finished` token to stop decoding. And then we can remove all stop tokens in the code.

**Comment by Rayhane-mamah** [comment]
Hello,

There are two points to discuss in this question:
First, we do need to train a stop token classifier to provide the model
with a dynamic stop option at test/synthesis times.
Second, yes that line of code can be dropped if we set impute_finished to
False. In the current state of the repo it is necessary. I did however
change this process (locally) with a loss mask which goes much faster. So
I'll probably remove the line you referenced in the next update.

It has been a while since I last pushed anything, many modifications are
occuring at the same time and I'm not finding much time to make them.
Thanks for you contribution however, excellent point!
